source/slang/slang-ir-simplify-cfg.cpp

```diff
@@ -123,7 +142,13 @@ static bool isTrivialSingleIterationLoop(
 
 static bool doesLoopHasSideEffect(IRGlobalValueWithCode* func, IRLoop* loopInst)
 {
-    auto blocks = collectBlocksInLoop(func, loopInst);
+    List<IRBlock*> multiLevelBreakBlocks;
+    auto blocks = collectBlocksInRegion(func, loopInst, multiLevelBreakBlocks);
```

Comment:
Are there any other places that require the actual break blocks? If not then we should just return a number instead of the whole list.